nats.go

```diff
@@ -0,0 +1,245 @@
+package gorillaz
+
+import (
+	"context"
+	"fmt"
+	"github.com/nats-io/nats.go"
+	"github.com/opentracing/opentracing-go"
+	"github.com/skysoft-atm/gorillaz/stream"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/proto"
+	"time"
+)
+
+// PullNatsStream returns the next subject and event for the given stream and consumer
+// If the event is processed successfully, it must be acknowledge to get a new message. If the message is not acknowledge, then PullNatsStream will return the same event multiple times
+// If no message is available after the ctx timeout, then an error nats.ErrTimeout is returned with an empty subject and an event nil
+func (g *Gaz) PullNatsStream(ctx context.Context, stream string, consumer string) (subject string, event *stream.Event, err error) {
+	subj := "$JS.API.CONSUMER.MSG.NEXT." + stream + "." + consumer
```

Comment:
wouldn't we need to create the durable consumer first?

Reply:
I assume the consumers and subscribers are already created.
I don't think it's a good idea to hide the creation and streams and consumers inside gorillaz, because we're going to bring all the complexity of https://github.com/nats-io/jsm.go inside
I'd rather start by letting the user use jsm.go, and once we're really sure how we create consumers, decide if we really want to hide it inside